src/proxy/outbound.rs

```diff
@@ -1009,6 +1012,105 @@ mod tests {
         .await;
     }
 
+    #[tokio::test]
+    async fn service_ip_families() {
+        initialize_telemetry();
+        let workload = XdsAddressType::Workload(XdsWorkload {
+            uid: "cluster1//v1/Pod/default/dual".to_string(),
+            addresses: vec![
+                Bytes::copy_from_slice(&[127, 0, 0, 2]),
+                Bytes::copy_from_slice("ff06::c3".parse::<Ipv6Addr>().unwrap().octets().as_slice()),
+            ],
+            tunnel_protocol: 1,
+            services: std::collections::HashMap::from([(
+                "/example.com".to_string(),
+                PortList {
+                    ports: vec![Port {
+                        service_port: 80,
+                        target_port: 1234,
+                    }],
+                },
+            )]),
+            ..Default::default()
+        });
+        let svc = |f: IpFamilies| {
+            let mut s = XdsService {
+                hostname: "example.com".to_string(),
+                addresses: vec![
+                    XdsNetworkAddress {
+                        network: "".to_string(),
+                        address: vec![127, 0, 0, 3],
+                    },
+                    XdsNetworkAddress {
+                        network: "".to_string(),
+                        address: "::3".parse::<Ipv6Addr>().unwrap().octets().into(),
+                    },
+                ],
+                ports: vec![
+                    Port {
+                        service_port: 80,
+                        target_port: 0, // named port
```

Comment:
wdym by named port?

Reply:
Named port means in the service.spec.ports.targetPort it is a name instead of a number. This was copy+pastyed from another test though, not needed on this one - I cleaned it up. Thanks